### DISCUSSION TITLE
New data viewer

### DISCUSSION THREAD

**Discussion by pierrethibault** [ptycho/ptypy#446] [Ideas]
One nice idea would be to have a html5 client for real-time data visualisation and additional interactions such as pausing, changing parameters on-the-fly, etc.

**Comment by pierrethibault** [comment]
Jypiter could be a nice option. This project: https://github.com/jupyter-incubator/declarativewidgets could be a way of having both python interactivity and nice responsive GUI. 

See also the video: https://www.youtube.com/watch?v=SJiezXPhVv8 (minutes 25-35 and around 45).

**Comment by daurer** [comment]
We could also try combining jupyter with bokeh:
http://bokeh.pydata.org/en/latest/docs/user_guide/notebook.html

**Comment by bjoernenders** [comment]
This idea is addressed by the ptyweb viewer https://github.com/ptycho/ptyweb

**Comment by Nordicus** [comment]
I think two features that might be nice for visualisations are

1. The option to include a scale bar on the real-space object reconstruction image. It's nice not to need another program to include scale information.
2. Making it easier to simultaneously visualise multiple reconstructions, perhaps as some sort of gif, and have the error metric between them compared.
